server/controllers/user.js

```diff
@@ -49,6 +49,42 @@ class User {
       token
     );
   }
+
+  /**
+   * Login a User
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} User Object
+   */
+  static async signin(request, response) {
+    const {
+      email,
+      password,
+    } = request.body;
+    const findUser = await userRepo.getUserByEmail(email);
+    if (findUser) {
+      const validatePassword = await compare(password, findUser.password);
+      if (validatePassword === false) {
+        return badHttpResponse(
+          response,
+          403,
+          'incorrect password, please try again'
+        );
+      }
+      const token = generateToken(findUser.id);
+      return goodHttpResponse(
+        response,
+        200,
+        `Hello ${findUser.username}, Welcome Back!`,
+        token
+      );
+    }
+    return badHttpResponse(
+      response,
+      404,
+      'User does not exist'
```

Comment:
I think the error message should also inform the user about the actions to take. Consider changing this to ```"This account does not exist. Consider sign up."```